ui/components/app/snaps/tx-insight-warnings/tx-insight-warnings.js

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  IconName,
+  AvatarIcon,
+  AvatarIconSize,
+  Text,
+  Checkbox,
+  Box,
+  Button,
+  ButtonSecondarySize,
+  ButtonPrimarySize,
+  ButtonVariant,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SnapUIRenderer } from '../snap-ui-renderer';
+import { DelineatorType } from '../../../../helpers/constants/snaps';
+import { stripHttpSchemes } from '../../../../helpers/utils/util';
+
+export default function TxInsightWarnings({
+  warnings,
+  type = 'confirming',
+  origin,
+  onCancel,
+  onSubmit,
+}) {
+  const t = useI18nContext();
+  // values represent if a warning is collapsed
+  const defaultWarningState = warnings.reduce((warningState, warning, idx) => {
+    if (idx === 0) {
+      warningState[warning.snapId] = false;
+    } else {
+      warningState[warning.snapId] = true;
+    }
+    return warningState;
+  }, {});
+
+  const [warningState, setWarningState] = useState(defaultWarningState);
+  const [isChecked, setIsChecked] = useState(false);
+
+  const handleOnChange = () => setIsChecked(!isChecked);
+
+  const handleWarningClick = (snapId) => {
+    const newState = { ...warningState };
+    const nextSnapState = !warningState[snapId];
+    const willClose = nextSnapState;
+    newState[snapId] = nextSnapState;
+    if (!willClose) {
+      const openWarningId = Object.keys(warningState).find(
+        (warningId) => warningId !== snapId && !warningState[warningId],
+      );
+      if (openWarningId !== undefined) {
+        newState[openWarningId] = true;
+      }
+    }
+    setWarningState(newState);
+  };
```

Comment:
It seems easier to just revert to the initial state (with the first item closed), and go from there?